app/src/main/java/de/xikolo/controllers/video/base/BaseVideoPlayerActivity.kt

```diff
@@ -153,6 +161,22 @@ abstract class BaseVideoPlayerActivity : BaseActivity(), VideoStreamPlayerFragme
         return hasCutout || aspectRatio - playerFragment.playerView.aspectRatio > 0.01
     }
 
+    override fun onToggleFullscreen(): Boolean {
+        val isPortrait: Boolean
+        requestedOrientation = if (isInLandscape()) {
+            isPortrait = true
+            ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        } else {
+            isPortrait = false
+            ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+        }
+        Handler().postDelayed({
+            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+        }, 3000) // user has 3 seconds to rotate device, otherwise it will change back
```

Comment:
This should force the orientation, no matter what orientation the device is actually in, just like the YouTube app does. This makes it possible to switch to fullscreen even when auto-rotate is off. This state is even maintained there so that when the app goes into the background (device back to portrait) and is opened again, the forced landscape orientation is restored. Have a look at the behavior of this app in detail, I find it really useful.

Reply:
Implemented rotation like in the YouTube app

Reply:
The experience is much better now! Unfortunately, I also haven't found a cleaner solution to implement this.

One last question: when auto-rotate is on, the user switches to full/small screen with the button and then rotates the device, the layout is flipped two times again. Can we avoid that?

Reply:
>One last question: when auto-rotate is on, the user switches to full/small screen with the button and then rotates the device, the layout is flipped two times again. Can we avoid that?

@bengelhaupt What is the status here?

Reply:
> > One last question: when auto-rotate is on, the user switches to full/small screen with the button and then rotates the device, the layout is flipped two times again. Can we avoid that?
> 
> @bengelhaupt What is the status here?

I can't reproduce that behavior on my device. Do you mean that after the screen has rotated through the button and then the device is rotated the same way, the screen flips back to normal and then into the rotated state?

Reply:
Exactly.

Reply:
Okay, I can reproduce it on the emulator. I will have a look onto it. Did it happen on a real device for you?

Reply:
Yes, on my Pixel 3.

Reply:
I rewrote the custom rotation detection. Can you test if this works on your device?

Reply:
Alright, now it works sometimes, still very sensitive. Anyway, I'm fine with it and I guess it is not worth it to optimize this further atm. Thank you!